src/credentials_callback.h

```diff
@@ -0,0 +1,44 @@
+/**
+ * \file   credentials_callback.h
+ * \author Lars Fröhlich
+ * \date   Created on January 16, 2024
+ * \brief  Declaration of the credentials callback function.
+ *
+ * \copyright Copyright 2024 Deutsches Elektronen-Synchrotron (DESY), Hamburg
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 2.1 of the license, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+#ifndef LIBGIT4CPP_CREDENTIALS_CALLBACK_H_
+#define LIBGIT4CPP_CREDENTIALS_CALLBACK_H_
+
+namespace git {
+extern "C" {
```

Comment:
I'm not sure this is a good idea (extern C within a namespace).

From C++ and when we compile the function name is `git::credentials_callback()`, but of course the symbol later on will be just `credentials_callback()` because it should be callable/linkable with C.

When some person uses libgit4cpp and creates a `my_very_own::credentials_callback` in a similar way we will end up with two identically named symbols.

If we do not put it in the namespace the (C++) compiler might warn earlier on.

I find the following explanation makes it most obvious.
The code is kind of misleading imho.

![image](https://github.com/taskolib/libgit4cpp/assets/16012374/0c0ad5b0-7b03-4f86-84ca-b3647200efda)

Reply:
> The code is kind of misleading imho.

I agree, the namespace does not make any sense. I'll rename the function to something ugly like `libgit4cpp_credentials_callback()`.